### PR TITLE
Feature/back 437 - create tests for UniswapV2 forks

### PR DESCRIPTION
**Pull request by Verisana**

### CODE REVIEW

File: src/dex/uniswap-v2/uniswap-v2-e2e-mainnet.test.ts [hunk 1]
```diff
@@ -0,0 +1,1333 @@
...
+      provider,
+    );
+  });
+
+  describe('UniswapV2', () => {
+    const dexKey = 'UniswapV2';
+    const network = Network.MAINNET;
+    const tokens = Tokens[Network.MAINNET];
+    const holders = Holders[Network.MAINNET];
+    const provider = new JsonRpcProvider(ProviderURL[network]);
```
Comment (ColonelJ) [inline src/dex/uniswap-v2/uniswap-v2-e2e-mainnet.test.ts:39]: don't need these 4 lines, the values are already set above...

**Review by ColonelJ — CHANGES_REQUESTED** [review]
There are a lot of tests in the mainnet UniswapV2 E2E which don't have a dexKey and therefore don't belong in dex-lib (can use any DEX).  I think they should be left in API's e2e tests, and removed from here.

File: src/dex/uniswap-v2/uniswap-v2-e2e-mainnet.test.ts (hunk 1 above)
Comment (Verisana) [reply src/dex/uniswap-v2/uniswap-v2-e2e-mainnet.test.ts:39]: Fixed

**Comment by Verisana** [comment]
@ColonelJ 
I agree. Removed from this tests all blocks on `Mainnet`:

```
ContractMethod: BuyMethod
ContractMethod: MegaPath
Multiswap Edge Cases
```
And two test cases on `Polygon` and `Mainnet` without clear `dexKey`

**Review by ColonelJ — APPROVED** [review]
